### DISCUSSION TITLE
qnap e caddy su docker

### DISCUSSION THREAD

**Discussion by g4box** [indomus/forum#2340] [GENERALE - Discussioni generiche]
Sto cercando di far funzionare la guida ufficiale
https://indomus.it/guide/installare-e-configurare-caddy-con-docker-su-qnap-via-containerstation/
Premetto di aver già cercato sul forum, ma ho trovato una sola discussione e non riporta il mio problema.
Cmq ahimè non ci riesco a farlo funzionare e devo chiedere aiuto! Ho aperto le porte 443 e 80 verso l'ip del nas e ricontrollando piu volte credo di aver configurato il container con i volumi e le porte in maniera corretta ma ottengo questo risultato. Allego il log

![Appunti01](https://user-images.githubusercontent.com/95856257/219011315-32479efe-3ca4-42a1-91d8-c7cf1965757e.jpg)

**Comment by snakuzzo** [comment]
Guarda il primo errore.
C'è un connection refused.
Probabilmente hai aperto correttamente le porte sul router, ma hai sbagliato a configurare l'esposizione delle porte del container

**Reply by g4box** [reply]
è si sono un nabbo non avevo replicato le porte in host. Sono arrivato al punto di ottenere i certificati definitivi, ho riconfigurato il Caddyfile e ho riconfigurato il configuration.ylm per HA aggiungendo nella sezione "html:" l'indirizzo ip del docker caddy come da guida, ma niente  ottengo HTTP ERROR 502. mentre se non aggiungo la porta ottengo regolarmente "saluti dal crew...". dove sbaglio?

**Reply by snakuzzo** [reply]
Devi postare tutta la configurazione dei container e del Caddyfile... 
Se ti dà 502 vuol dire che Caddy non riesce a contattare l'origin (che sarebbe ip:porta di HA)

**Reply by g4box** [reply]
snakuzzo funziona! Ho impostato sul file di configurazione di HA tutta la classe di indirizzi docker anzichè il solo ip del container di caddy... ho riavviato direttamente il container da container station (non ho riavviato da home assistant come prima forse è per questo) e magicamente si è collegato in connessione sicura. Grazie per il supporto e per questa guida.
